attrd/commands.c

```diff
@@ -982,14 +982,14 @@ attrd_cib_callback(xmlNode * msg, int call_id, int rc, xmlNode * output, void *u
 }
 
 void
-write_attributes(bool all, bool peer_discovered)
+write_attributes(bool all)
 {
     GHashTableIter iter;
     attribute_t *a = NULL;
 
     g_hash_table_iter_init(&iter, attributes);
     while (g_hash_table_iter_next(&iter, NULL, (gpointer *) & a)) {
-        if (peer_discovered && a->unknown_peer_uuids) {
+        if (!all && a->unknown_peer_uuids) {
```

Comment:
Wait. What??  Thats not a redundant argument.

Reply:
If we're writing out all attributes, we don't care about whether a peer has been discovered (since we're writing them all out regardless). If we're not writing out all attributes, we're writing out attributes with previously unknown peer uuids. Which explains why peer_discovered == !all for all callers.